src/firebase/CRUDServices/user.js

```diff
@@ -0,0 +1,114 @@
+export default class User {
+    /**
+     * Private fields for user object.
+     */
+    #id;
+    #username;
+    #password;
+    #firstName;
+    #lastName;
+    #email;
+    #role;
+    /**
+     * A constructor to create a User object.
+     *
+     * @param {String} email email of the user.
+     * @param {String} password password of the user.
+     */
+    constructor(email, password) {
+      this.#email = email;
+      this.#password = password;
+    }
+    /**
+     * Setter for id of the user.
+     * @param {int} id user id of the user.
+     */
+    set id(id) {
+      this.#id = id;
+    }
+    /**
+     * Return the id of this user.
+     */
+    get id() {
+      return this.#id;
+    }
+    /**
+     * Setter for username of the user.
+     * @param {String} username username of the user.
+     */
+    set username(username) {
+      this.#username = username;
+    }
+    /**
+     * Return the username of this user.
+     */
+    get username() {
+      return this.#username;
+    }
+    /**
+     * Setter for password of the user.
+     * @param {String} password password of the user.
+     */
+    set password(password) {
+      this.#password = password;
+    }
+    /**
+     * Return the password of this user.
+     */
+    get password() {
+      return this.#password;
+    }
+    /**
+     * Setter for firstName of the user.
+     * @param {String} firstName firstName of the user.
+     */
+    set firstName(firstName) {
+      this.#firstName = firstName;
+    }
+    /**
+     * Return the firstName of this user.
+     */
+    get firstName() {
+      return this.#firstName;
+    }
+    /**
+     * Setter for lastName of the user.
+     * @param {String} lastName lastName of the user.
+     */
+    set lastName(lastName) {
+      this.#lastName = lastName;
+    }
+    /**
+     * Return the lastName of this user.
+     */
+    get lastName() {
+      return this.#lastName;
+    }
+    /**
+     * Setter for email of the user.
+     * @param {String} email email of the user.
+     */
+    set email(email) {
+      this.#email = email;
+    }
+    /**
+     * Return the email of this user.
+     */
+    get email() {
+      return this.#email;
+    }
+    /**
+     * Setter for role of the user.
+     * @param {int} role user role of the user. 0 represents participants.
```

Comment:
Are you suggesting just using string like 'participant' or 'admin'? I just thought by using string literal, typo or upper lower case may be a problem? Thought of using enum but after I searched about it, there is not a good way to do so like Java? (maybe I am just not good at finding things out on JavaScript.) 
@bleachalon What do you think?

Reply:
I think you both have a valid point. If you decide to use an integer, I would suggest changing the name to a more semantic one such as permission level or authorization level instead of role.

Reply:
I will revise them all together after hearing back from your meeting tomorrow.